Library/Homebrew/sandbox.rb

```diff
@@ -89,6 +105,53 @@ def deny_write_homebrew_repository
     end
   end
 
+  def deny_read_shell_files
+    # Configuration.
+    # deny_read "/Users/#{ENV["USER"]}/.bash_profile"
+    # deny_read "/Users/#{ENV["USER"]}/.cshrc"
+    # deny_read "/Users/#{ENV["USER"]}/.config/fish/config.fish"
+    # deny_read "/Users/#{ENV["USER"]}/.kshrc"
+    # deny_read "/Users/#{ENV["USER"]}/.tcshrc"
+    # deny_read "/Users/#{ENV["USER"]}/.zshrc"
+
+    # Maybe this way is cleaner, although less explicit.
+    # Should this use a "/[^/]+/" regex instead of explicit USER to ensure
+    # we're not just protecting the active user on the machine but all of them?
+    %w[bash_profile cshrc config/fish/config.fish kshrc tcshrc zshrc].each do |s|
+      deny_read "/Users/#{ENV["USER"]}/.#{s}"
+    end
+
+    # Same as above. Seek review on preference.
+    # History.
+    # deny_read "/Users/#{ENV["USER"]}/.bash_history"
+    # deny_read "/Users/#{ENV["USER"]}/.zsh_history"
+    # deny_read "/Users/#{ENV["USER"]}/.config/fish/fish_history"
+    # deny_read "/Users/#{ENV["USER"]}/.sh_history"
+    # deny_read "/Users/#{ENV["USER"]}/.history"
+    %w[bash_history zsh_history config/fish/fish_history sh_history history].each do |s|
```

Comment:
The location of the history files for most of these shells is configurable; we should probably respect that.

Should probably check `$HISTFILE`, in case the user has a custom history file location. And `$ZDOTDIR` for `zsh` to cover the case of custom startup file location. (I think just `$HISTFILE` might cover it for `bash` and `zsh`.

As of `fish` 2.3.0, its history file is in `~/.local/share/fish` or under `$XDG_DATA_HOME` instead of under `~/.config` or `$XDG_CONFIG_HOME`. I don't see a `fish` variable to get the location of the history file directly.